sys/netbsd/sys.txt

```diff
@@ -190,6 +194,46 @@ ucred {
 	gid	gid
 }
 
+posix_spawn_file_actions {
+	size	int32
+	len	int32
+	fae	ptr[in, posix_spawn_file_actions_entry]
+}
+
+posix_spawn_file_actions_entry {
+	fae_action	int32
+	fae_filedes	int32
+	fae_data	fae_data_t
```

Comment:
fae_action/filedes/data seem to be interrelated. To make the description more precise we need to move the union to the upper level. Something along the following lines:

```
posix_spawn_file_actions_entry [
  open posix_spawn_file_actions_entry_t[FAE_OPEN, const[0, int32], open_t]
  dup posix_spawn_file_actions_entry_t[FAE_DUP2, fd, const[0, int32]]
  close posix_spawn_file_actions_entry_t[FAE_CLOSE, fd, void]
]

type posix_spawn_file_actions_entry_t[ACTION, FILEDES, DATA] {
  fae_action const[ACTION, int32]
  fae_filedes FILEDES
  fae_data DATA
}
```

Reply:
Um.. so I pass the union to the function? won't that make the fae_action be a object of a structure of a union?
I am not very clear about this.

Reply:
We pass a pointer to a posix_spawn_file_actions to the function.
posix_spawn_file_actions contains a pointer to an array of posix_spawn_file_actions_entry unions.

> won't that make the fae_action be a object of a structure of a union?

I don't understand what you mean. Please rephrase.

Reply:
According to the code - we access fae_action as 
posix_spawn_file_actions_entry.fae_action 

According to the modifications you suggested won't it become
posix_spawn_file_actions_entry.open.fae_action

Reply:
> According to the modifications you suggested won't it become
posix_spawn_file_actions_entry.open.fae_action

The syzkaller names don't matter, only the resulting binary layout matters. If we wrap a struct in another struct or a union, the memory layout does not change.

Reply:
ping
do you have any other questions or stuck somewhere?
we are close

Reply:
I will try to wrap this one up by tmrw. Was caught up with something else